lib/scholar/cluster/k_means.ex

```diff
@@ -253,27 +259,37 @@ defmodule Scholar.Cluster.KMeans do
       |> Nx.as_type({:s, 8})
       |> Nx.argmax(tie_break: :low, axis: 1)
 
-    Nx.take(x, idx)
+    {Nx.take(x, idx), new_key}
   end
 
-  defnp k_means_plus_plus(x, num_clusters, num_runs) do
+  deftransformp check_seed(seed) do
+    case seed do
+      nil -> :erlang.system_time()
+      _ -> seed
+    end
+  end
+
+  defnp k_means_plus_plus(x, num_clusters, num_runs, opts) do
     inf = Nx.Constants.infinity()
     {num_samples, num_features} = Nx.shape(x)
     centroids = Nx.broadcast(inf, {num_runs, num_clusters, num_features})
     inertia = Nx.broadcast(0.0, {num_runs, num_samples})
 
-    first_centroid_idx = Nx.random_uniform({num_runs}, 0, num_samples - 1)
+    seed = check_seed(opts[:random_seed])
```

Comment:
I would compute the seed on `deftransform fit`. Otherwise, once this function is compiled once for a given shape, the seed will never change.